File: app/src/main/java/com/worldventures/dreamtrips/wallet/analytics/firmware/action/ViewSdkUpdateAction.java
package com.worldventures.dreamtrips.wallet.analytics.firmware.action;

import android.support.annotation.NonNull;

import com.worldventures.dreamtrips.api.smart_card.firmware.model.FirmwareInfo;
import com.worldventures.dreamtrips.core.utils.tracksystem.AdobeTracker;
import com.worldventures.dreamtrips.core.utils.tracksystem.AnalyticsEvent;
import com.worldventures.dreamtrips.core.utils.tracksystem.Attribute;
import com.worldventures.dreamtrips.wallet.domain.entity.FirmwareUpdateData;

@AnalyticsEvent(action = "wallet:SmartCard Update:Step 1",
                navigationState = true,
                trackers = AdobeTracker.TRACKER_KEY)
public class ViewSdkUpdateAction extends FirmwareAnalyticsAction {

   @Attribute("scupdatestep1") final String updateStep = "1";
   @Attribute("currentversion") String currentVersion;
   @Attribute("latestversion") String latestVersion;
   @Attribute("dtupdaterqrd") String updateRequired;

   @Override
   public void setFirmwareData(@NonNull FirmwareUpdateData data) {
      super.setFirmwareData(data);
      FirmwareInfo info = data.firmwareInfo();
      if (info != null) {
         this.latestVersion = info.firmwareVersion();
         this.updateRequired = info.isCompatible() ? "Yes" : "No";
      }
      this.currentVersion = data.currentFirmwareVersion().nordicAppVersion();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/analytics/firmware/action/ViewSdkUpdateAction.kt
package com.worldventures.wallet.analytics.firmware.action

import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.core.service.analytics.Attribute
import com.worldventures.janet.analytics.AnalyticsEvent
import com.worldventures.wallet.domain.entity.FirmwareUpdateData

@AnalyticsEvent(action = "wallet:SmartCard Update:Step 1", navigationState = true, trackers = arrayOf(AdobeTracker.TRACKER_KEY))
class ViewSdkUpdateAction : FirmwareAnalyticsAction() {

   @Attribute("scupdatestep1") internal val updateStep = "1"
   @Attribute("currentversion") internal lateinit var currentVersion: String
   @Attribute("latestversion") internal var latestVersion: String? = null
   @Attribute("dtupdaterqrd") internal var updateRequired: String? = null

   override fun setFirmwareData(data: FirmwareUpdateData) {
      super.setFirmwareData(data)
      val info = data.firmwareInfo
      if (info != null) {
         this.latestVersion = info.firmwareVersion()
         this.updateRequired = if (info.isCompatible) "Yes" else "No"
      }
      this.currentVersion = data.currentFirmwareVersion.nordicAppVersion
   }
}
